packages/patternfly-3/react-console/src/DesktopViewer/ConnectWithRemoteViewer.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import MoreInformationContent from './MoreInformationContent';
+import { generateVVFile, downloadFile } from './vvFileGenerator';
+import consoleDetailPropType from './consoleDetailPropType';
+
+class MoreInformation extends React.Component {
+  state = { expanded: false };
+
+  onClick = e => {
+    e.preventDefault();
+    this.setState(prevState => ({ expanded: !prevState.expanded }));
+  };
+
+  render() {
+    return (
+      <div>
+        <div className="remote-viewer-more-info">
+          <a href="#" tabIndex="0" onClick={this.onClick}>
+            {this.state.expanded ? <span className="fa fa-angle-down" /> : <span className="fa fa-angle-right" />}
+            &nbsp; More Information
+          </a>
+        </div>
+        {this.state.expanded && this.props.children}
+      </div>
+    );
+  }
+}
+MoreInformation.propTypes = {
+  children: PropTypes.any.isRequired
+};
+MoreInformation.defaultProps = {};
+
+const ConnectWithRemoteViewer = ({ children, onGenerate, onDownload, spice, vnc }) => {
+  const onClick = () => {
+    const console = spice || vnc; // strictly prefer spice over vnc
+    const type = spice ? 'spice' : 'vnc';
```

Comment:
use constants?

Reply:
done. Newly using same constants as in #562.
So far copy&paste since #562 is not yet merged. I will refactor asap.